pulp_2to3_migrate/app/pre_migration.py

```diff
@@ -181,20 +179,21 @@
     Pre-migrate a pulp 2 repo.
 
     Args:
-        record(dict): Pulp 2 repository data
+        record(Repository): Pulp 2 repository data
 
     Return:
         repo(Pulp2Repository): A pre-migrated repository
     """
-    last_unit_added = (record['last_unit_added'] and
-                       timezone.make_aware(record['last_unit_added'], timezone.utc))
-    last_unit_removed = (record['last_unit_removed'] and
-                         timezone.make_aware(record['last_unit_removed'], timezone.utc))
+
+    last_unit_added = (record.last_unit_added and
+                       timezone.make_aware(record.last_unit_added, timezone.utc))
+    last_unit_removed = (record.last_unit_removed and
+                         timezone.make_aware(record.last_unit_removed, timezone.utc))
 
     # repo is mutable, it needs to be created or updated
     repo, created = Pulp2Repository.objects.update_or_create(
-        pulp2_object_id=record['id'],
-        pulp2_repo_id=record['repo_id'],
+        pulp2_object_id=record.id,
+        pulp2_repo_id=record.repo_id,
```

Comment:
Thank you for all the changes!
Please fix the `description` below as well. I guess it was left unnoticed because of rebase (you added it recently)